src/wasm.h

```diff
@@ -2255,11 +2255,21 @@ enum class ModuleItemKind {
 
 class Export {
 public:
-  // exported name - note that this is the key, as the internal name is
-  // non-unique (can have multiple exports for an internal, also over kinds)
+  // exported name - note that this is the key, as the internal name,
+  // or the exported type, is non-unique (can have multiple exports for an
+  // internal, also over kinds)
   Name name;
-  Name value; // internal name
   ExternalKind kind;
+
+private:
+  std::variant<Name, HeapType> value; // internal name or exported type
+
+public:
+  Export(Name name, ExternalKind kind, std::variant<Name, HeapType> value)
+    : name(name), kind(kind), value(value) {
+    assert(std::get_if<Name>(&value));
+  }
+  Name* getInternalName() { return std::get_if<Name>(&value); }
```

Comment:
How about this?
```suggestion
  Name& getInternalName() { return std::get<Name>(&value); }
```
That seems nicer (avoiding a pointer) and it would also check it is the right variant.

Reply:
Hmm, true. Though that usage seems very rare in this PR. That makes me wonder if it would be nicer to do what `Module` does and have `getInternalName()` vs `getInternalNameOrNull()`.

Though in other places we do otherwise, and this is consistent with that (as I see was noted before).

But it feels risky in the current form, as all the `*getInternalName()` are not actually checked - C++ will just have undefined behavior there. That makes me lean to the other direction.

Reply:
Dereferencing a null pointer may be UB, but at least it's usually explosive UB. And we do have ASan on CI that would pick up most problems this could cause.

Reply:
You can look at the [second commit](26cf26427d08aa02938afb172dfb57c332bb45c5) to see what the change would look like.

Reply:
@tlively Yeah, I suppose it isn't a subtle breakage...

I see, thanks @vouillon 

Well, I don't feel strongly given those, and the difference is not large. We can always refactor later if we see this causes issues. Having `getInternalName` as an API makes that easier at least compared to before when it was just a field.